source/Ship.cpp

```diff
@@ -692,16 +665,28 @@ void Ship::FinishLoading(bool isNewInstance)
 	if(isNewInstance)
 		Recharge(true);
 	
-	// Add a default "launch effect" to any internal bays if this ship is crewed (i.e. pressurized).
-	for(Bay &bay : bays)
+	// Ensure that all defined bays are of a valid category. Remove and warn about any
+	// invalid bays. Add a default "launch effect" to any remaining internal bays if
+	// this ship is crewed (i.e. pressurized).
+	string warning;
+	for(auto it = bays.begin(); it != bays.end(); ++it)
+	{
+		Bay &bay = *it;
+		if(!GameData::CategorySet("bay type").count(bay.category))
+		{
+			bays.erase(it);
+			--it;
```

Comment:
`--it` does bad things when at the front of the vector. You'll need to have an empty 3rd statement in the for-loop initialization to avoid the double-increment when erasing.
```suggestion
			it = bays.erase(it);
```